google/cloud/firestore_v1/async_transaction.py

```diff
@@ -41,6 +52,10 @@
     from google.cloud.firestore_v1.query_profile import ExplainOptions
 
 
+T = TypeVar("T")
+P = ParamSpec("P")
```

Comment:
It looks like ParamSpec was added in python 3.10, and we need to support 3.8

Reply:
I added `typing_extensions` to support expanded type hinting for older python versions. Let me know if that works, otherwise I'll come up with something more hacky for the signatures.

Reply:
Sorry, adding an extra dependency to setup.py isn't an option unfortunately

Let me know if you're able to experiment with getting this working, or I can try to take it from here if you don't have the capacity

Thanks!

Reply:
I'm not going to have time to continue looking at this for the forseeable future.